### PR TITLE
[WIP][FLINK-25537]  [JUnit5 Migration] Module: flink-core

### PR DESCRIPTION
**Pull request by GOODBOY008**
## What is the purpose of the change

*Update the flink-core module to AssertJ and JUnit 5 following the [JUnit 5 Migration Guide](https://docs.google.com/document/d/1514Wa_aNB9bJUen4xm5uiuXOooOJTtXqS_Jqk9KJitU/edit)
Also it touches tests extending from org.apache.flink.table.catalog.CatalogTest*

## Brief change log

*Use JUnit5 and AssertJ in tests instead of JUnit4 and Hamcrest*

## Verifying this change

This change is a code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-core/src/test/java/org/apache/flink/api/common/accumulators/AverageAccumulatorTest.java [hunk 1]
```diff
@@ -94,20 +93,20 @@ public void testMergeSuccess() {
...
-    public void testMergeFailed() {
+    void testMergeFailed() {
         AverageAccumulator average = new AverageAccumulator();
         Accumulator<Double, Double> averageNew = null;
         average.add(1);
         try {
             average.merge(averageNew);
             fail("should fail with an exception");
         } catch (IllegalArgumentException e) {
-            assertNotNull(e.getMessage());
```
Comment (reswqa) [inline flink-core/src/test/java/org/apache/flink/api/common/accumulators/AverageAccumulatorTest.java:109]: Might replace this code block by assertThatThrownBy

**Comment by snuyanzin** [comment]
Hi @GOODBOY008 in case you are still willing to continue with this PR
https://github.com/apache/flink/pull/19667 was merged. It should significantly ease migration of `SerializerTestBase ` successors.
So makes sense to rebase.

**Comment by GOODBOY008** [comment]
> Hi @GOODBOY008 in case you are still willing to continue with this PR #19667 was merged. It should significantly ease migration of `SerializerTestBase ` successors. So makes sense to rebase.

Ok, I  will rebase my branch.

**Comment by GOODBOY008** [comment]
@flinkbot run azure

File: flink-core/src/test/java/org/apache/flink/api/common/accumulators/AverageAccumulatorTest.java [hunk 2]
```diff
@@ -94,21 +92,21 @@ public void testMergeSuccess() {
...
-            assertTrue(
-                    e.getMessage().contains("The merged accumulator must be AverageAccumulator."));
+            assertThat(e.getMessage()).isNotNull();
+            assertThat(e.getMessage())
+                    .contains("The merged accumulator must be AverageAccumulator.");
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/accumulators/AverageAccumulatorTest.java:109]: We can use `assertThatThrows` to simplify.

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 3]
```diff
@@ -44,12 +39,12 @@ public void singleImmediateWatermark() {
 
         watermarkOutput.emitWatermark(new Watermark(0));
 
-        assertThat(underlyingWatermarkOutput.lastWatermark(), is(watermark(0)));
-        assertThat(underlyingWatermarkOutput.isIdle(), is(false));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isEqualTo(watermark(0));
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(false);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:43]:
```suggestion
        assertThat(underlyingWatermarkOutput.isIdle()).isFalse();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 4]
```diff
@@ -58,29 +53,29 @@ public void singleImmediateIdleness() {
 
         watermarkOutput.markIdle();
 
-        assertThat(underlyingWatermarkOutput.lastWatermark(), nullValue());
-        assertThat(underlyingWatermarkOutput.isIdle(), is(true));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isNull();
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(true);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:57]:
```suggestion
        assertThat(underlyingWatermarkOutput.isIdle()).isTrue();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 5]
```diff
@@ -58,29 +53,29 @@ public void singleImmediateIdleness() {
...
+    void singleImmediateWatermarkAfterIdleness() {
         TestingWatermarkOutput underlyingWatermarkOutput = createTestingWatermarkOutput();
         WatermarkOutputMultiplexer multiplexer =
                 new WatermarkOutputMultiplexer(underlyingWatermarkOutput);
 
         WatermarkOutput watermarkOutput = createImmediateOutput(multiplexer);
 
         watermarkOutput.markIdle();
-        assertThat(underlyingWatermarkOutput.isIdle(), is(true));
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(true);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:69]:
```suggestion
        assertThat(underlyingWatermarkOutput.isIdle()).isTrue();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 6]
```diff
@@ -58,29 +53,29 @@ public void singleImmediateIdleness() {
...
 
         watermarkOutput.emitWatermark(new Watermark(0));
 
-        assertThat(underlyingWatermarkOutput.lastWatermark(), is(watermark(0)));
-        assertThat(underlyingWatermarkOutput.isIdle(), is(false));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isEqualTo(watermark(0));
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(false);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:74]:
```suggestion
        assertThat(underlyingWatermarkOutput.isIdle()).isFalse();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 7]
```diff
@@ -93,12 +88,12 @@ public void multipleImmediateWatermark() {
         watermarkOutput2.emitWatermark(new Watermark(5));
         watermarkOutput3.markIdle();
 
-        assertThat(underlyingWatermarkOutput.lastWatermark(), is(watermark(2)));
-        assertThat(underlyingWatermarkOutput.isIdle(), is(false));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isEqualTo(watermark(2));
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(false);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:92]:
```suggestion
        assertThat(underlyingWatermarkOutput.isIdle()).isFalse();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 8]
```diff
@@ -145,19 +140,19 @@ public void combinedWatermarkDoesNotRegressWhenIdleOutputRegresses() {
...
         TestingWatermarkOutput underlyingWatermarkOutput = createTestingWatermarkOutput();
         WatermarkOutputMultiplexer multiplexer =
                 new WatermarkOutputMultiplexer(underlyingWatermarkOutput);
 
         multiplexer.onPeriodicEmit();
 
-        assertThat(underlyingWatermarkOutput.lastWatermark(), is(nullValue()));
-        assertThat(underlyingWatermarkOutput.isIdle(), is(false));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isEqualTo(nullValue());
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(false);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:151]:
```suggestion
        assertThat(underlyingWatermarkOutput.isIdle()).isFalse();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 9]
```diff
@@ -186,12 +181,12 @@ public void singleDeferredWatermark() {
         watermarkOutput.emitWatermark(new Watermark(0));
         multiplexer.onPeriodicEmit();
 
-        assertThat(underlyingWatermarkOutput.lastWatermark(), is(watermark(0)));
-        assertThat(underlyingWatermarkOutput.isIdle(), is(false));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isEqualTo(watermark(0));
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(false);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:185]:
```suggestion
        assertThat(underlyingWatermarkOutput.isIdle()).isFalse();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 10]
```diff
@@ -201,12 +196,12 @@ public void singleDeferredIdleness() {
         watermarkOutput.markIdle();
         multiplexer.onPeriodicEmit();
 
-        assertThat(underlyingWatermarkOutput.lastWatermark(), nullValue());
-        assertThat(underlyingWatermarkOutput.isIdle(), is(true));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isNull();
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(true);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:200]:
```suggestion
        assertThat(underlyingWatermarkOutput.isIdle()).isTrue();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 11]
```diff
@@ -216,17 +211,17 @@ public void singleDeferredWatermarkAfterIdleness() {
         watermarkOutput.markIdle();
         multiplexer.onPeriodicEmit();
 
-        assertThat(underlyingWatermarkOutput.isIdle(), is(true));
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(true);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:214]:
```suggestion
        assertThat(underlyingWatermarkOutput.isIdle()).isTrue();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 12]
```diff
@@ -216,17 +211,17 @@ public void singleDeferredWatermarkAfterIdleness() {
...
 
         watermarkOutput.emitWatermark(new Watermark(0));
         multiplexer.onPeriodicEmit();
 
-        assertThat(underlyingWatermarkOutput.lastWatermark(), is(watermark(0)));
-        assertThat(underlyingWatermarkOutput.isIdle(), is(false));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isEqualTo(watermark(0));
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(false);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:220]:
```suggestion
        assertThat(underlyingWatermarkOutput.isIdle()).isFalse();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 13]
```diff
@@ -256,15 +251,15 @@ public void immediateUpdatesTakeDeferredUpdatesIntoAccount() {
 
         deferredOutput.emitWatermark(new Watermark(5));
 
-        assertThat(underlyingWatermarkOutput.lastWatermark(), is(nullValue()));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isEqualTo(nullValue());
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:254]:
```suggestion
        assertThat(underlyingWatermarkOutput.lastWatermark()).isNull();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 14]
```diff
@@ -241,12 +236,12 @@ public void multipleDeferredWatermark() {
 
         multiplexer.onPeriodicEmit();
 
-        assertThat(underlyingWatermarkOutput.lastWatermark(), is(watermark(2)));
-        assertThat(underlyingWatermarkOutput.isIdle(), is(false));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isEqualTo(watermark(2));
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(false);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:240]:
```suggestion
        assertThat(underlyingWatermarkOutput.isIdle()).isFalse();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 15]
```diff
@@ -301,11 +296,11 @@ public void lowerImmediateUpdateOnSameOutputDoesNotEmitCombinedUpdate() {
         deferredOutput.emitWatermark(new Watermark(5));
         immediateOutput.emitWatermark(new Watermark(2));
 
-        assertThat(underlyingWatermarkOutput.lastWatermark(), is(nullValue()));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isEqualTo(nullValue());
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:299]:
```suggestion
        assertThat(underlyingWatermarkOutput.lastWatermark()).isNull();
```

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java [hunk 16]
```diff
@@ -395,8 +393,8 @@ public void testNotEmittingIdleAfterAllSplitsRemoved() {
         multiplexer.unregisterOutput(id);
 
         multiplexer.onPeriodicEmit();
-        assertThat(underlyingWatermarkOutput.lastWatermark(), equalTo(emittedWatermark));
-        assertThat(underlyingWatermarkOutput.isIdle(), equalTo(false));
+        assertThat(underlyingWatermarkOutput.lastWatermark()).isEqualTo(emittedWatermark);
+        assertThat(underlyingWatermarkOutput.isIdle()).isEqualTo(false);
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexerTest.java:397]: same as above.

File: flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkStrategyTest.java [hunk 17]
```diff
@@ -164,19 +160,15 @@ public void testWithWatermarkAlignment() {
...
-                instanceOf(WatermarksWithIdleness.class));
+        assertThat(alignmentParameters.getWatermarkGroup()).isEqualTo(watermarkGroup);
+        assertThat(equalTo(maxAllowedWatermarkDrift.toMillis()))
+                .isEqualTo(alignmentParameters.getMaxAllowedWatermarkDrift());
+        assertThat(equalTo(WatermarksWithWatermarkAlignment.DEFAULT_UPDATE_INTERVAL.toMillis()))
+                .isEqualTo(alignmentParameters.getUpdateInterval());
+        assertThat(instanceOf(RecordTimestampAssigner.class))
+                .isEqualTo(strategy.createTimestampAssigner(assignerContext()));
+        assertThat(instanceOf(WatermarksWithIdleness.class))
+                .isEqualTo(strategy.createWatermarkGenerator(generatorContext()));
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/eventtime/WatermarkStrategyTest.java:171]: `equalTo` can be removed.

File: flink-core/src/test/java/org/apache/flink/api/common/functions/util/RuntimeUDFContextTest.java [hunk 18]
```diff
@@ -95,36 +93,36 @@ public void testBroadcastVariableSimple() {
...
+            assertThat(ctx.hasBroadcastVariable("name1")).isTrue();
+            assertThat(ctx.hasBroadcastVariable("name2")).isTrue();
 
             List<Integer> list1 = ctx.getBroadcastVariable("name1");
             List<Double> list2 = ctx.getBroadcastVariable("name2");
 
-            assertEquals(Arrays.asList(1, 2, 3, 4), list1);
-            assertEquals(Arrays.asList(1.0, 2.0, 3.0, 4.0), list2);
+            assertThat(list1).isEqualTo(Arrays.asList(1, 2, 3, 4));
+            assertThat(list2).isEqualTo(Arrays.asList(1.0, 2.0, 3.0, 4.0));
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/functions/util/RuntimeUDFContextTest.java:103]:
```suggestion
            assertThat(list1).containsExactly(1, 2, 3, 4);
            assertThat(list2).containsExactly(1.0, 2.0, 3.0, 4.0);
```

File: flink-core/src/test/java/org/apache/flink/api/common/io/DelimitedInputFormatSamplingTest.java [hunk 19]
```diff
@@ -72,7 +75,7 @@ public class DelimitedInputFormatSamplingTest {
     //  Setup
     // ========================================================================
 
-    @BeforeClass
+    @BeforeAll
     public static void initialize() {
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/io/DelimitedInputFormatSamplingTest.java:79]:
```suggestion
    static void initialize() {
```

File: flink-core/src/test/java/org/apache/flink/api/common/io/EnumerateNestedFilesTest.java [hunk 20]
```diff
@@ -26,15 +26,17 @@
...
 
 import java.io.File;
 import java.io.IOException;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
 public class EnumerateNestedFilesTest {
 
     @Rule public TemporaryFolder tempFolder = new TemporaryFolder();
```
Comment (Jiabao-Sun) [inline flink-core/src/test/java/org/apache/flink/api/common/io/EnumerateNestedFilesTest.java:42]:
```suggestion
    @TempDir private Path temporaryFolder;
```

**Review by Jiabao-Sun — COMMENTED** [review]
Thanks @GOODBOY008 for this contribution.
I left some comments.

**Pull request closed without merging**